### PR TITLE
add error handling to promises

### PR DESCRIPTION
**Pull request by karthikjeeyar**
This PR contains the following changes: 

* Fixed the code by adding error handling.
* Removed the eslint disable comment `// eslint-disable-next-line promise/catch-or-return`

**Trigger Pipeline error:**

screencast-localhost-9000-2019 07 17-16-27-42

**Trigger pipelinerun error:**

screencast-localhost-9000-2019 07 17-16-28-16

Fixes: https://jira.coreos.com/browse/ODC-1124

### CODE REVIEW

File: frontend/packages/dev-console/src/components/pipelines/PipelineDetailsPage.tsx [hunk 1]
```diff
@@ -12,33 +13,48 @@ import { PipelineRunModel, PipelineModel } from '../../models';
...
+              ],
+            });
+          })
+          .catch((error) => this.setState({ errorCode: error.response.status }));
+      })
+      .catch((error) => this.setState({ errorCode: error.response.status }));
   }
 
   render() {
+    if (this.state.errorCode && this.state.errorCode === 404) {
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/pipelines/PipelineDetailsPage.tsx]:
```suggestion
    if (this.state.errorCode === 404) {
```

File: frontend/packages/dev-console/src/components/pipelineruns/PipelineRunVisualization.tsx [hunk 2]
```diff
@@ -19,23 +20,30 @@ export class PipelineRunVisualization extends React.Component<
...
+      .then((res) => {
+        this.setState({
+          pipeline: res,
+        });
+      })
+      .catch((error) => this.setState({ errorCode: error.response.status }));
   }
 
   render() {
+    if (this.state.errorCode && this.state.errorCode === 404) {
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/pipelineruns/PipelineRunVisualization.tsx]:
```suggestion
    if (this.state.errorCode === 404) {
```

Comment (karthikjeeyar) [reply frontend/packages/dev-console/src/components/pipelineruns/PipelineRunVisualization.tsx]: updated

**Pull request merged**